src/types.ts

```diff
@@ -391,37 +391,70 @@ export interface NetworkConnectionResponse {
     status: string;
     remoteNetworkId: NetworkId;
     localNetworkId: NetworkId;
-    routingPolicy?: RoutingPolicy;
+    routingPolicy?: NetworkConnectionRoutingPolicy;
+}
+
+export interface NetworkIdResponse {
+    id: string;
+    name: string;
+    isDiscoverable: boolean;
+    routingPolicy?: NetworkIdRoutingPolicy;
 }
 
 interface NetworkId {
     id: string;
     name: string;
 }
 
-export interface RoutingPolicy {
-    crypto?: RoutingDest;
-    sen?: RoutingDest;
-    signet?: RoutingDest;
-    sen_test?: RoutingDest;
-    signet_test?: RoutingDest;
+export interface CustomCryptoRoutingDest {
+    scheme: NetworkScheme.CUSTOM;
+    dstType: NetworkDestType.EXCHANGE_ACCOUNT | NetworkDestType.VAULT_ACCOUNT;
+    dstId: string;
 }
 
-export interface RoutingDest {
-    scheme: Scheme;
-    dstType: NetworkDestType;
+export interface CustomFiatRoutingDest {
+    scheme: NetworkScheme.CUSTOM;
+    dstType: NetworkDestType.FIAT_ACCOUNT;
     dstId: string;
 }
 
-export enum Scheme {
-    AUTO = "AUTO",
+export interface DefaultNetworkRoutingDest {
+    scheme: NetworkScheme.DEFAULT;
+}
+
+export interface AutoNetworkRoutingDest {
+    scheme: NetworkScheme.AUTO;
+}
+
+export interface NoneNetworkRoutingDest {
+    scheme: NetworkScheme.NONE;
+}
+
+export interface NetworkConnectionRoutingPolicy {
+    crypto?: CustomCryptoRoutingDest | DefaultNetworkRoutingDest | NoneNetworkRoutingDest;
+    sen?: CustomFiatRoutingDest | DefaultNetworkRoutingDest | NoneNetworkRoutingDest;
+    signet?: CustomFiatRoutingDest | DefaultNetworkRoutingDest | NoneNetworkRoutingDest;
+    sen_test?: CustomFiatRoutingDest | DefaultNetworkRoutingDest | NoneNetworkRoutingDest;
+    signet_test?: CustomFiatRoutingDest | DefaultNetworkRoutingDest | NoneNetworkRoutingDest;
+}
+
+export interface NetworkIdRoutingPolicy {
+    crypto?: CustomCryptoRoutingDest | AutoNetworkRoutingDest | NoneNetworkRoutingDest;
+    sen?: CustomFiatRoutingDest | AutoNetworkRoutingDest | NoneNetworkRoutingDest;
```

Comment:
@yuval-fireblocks what do u think about group all of them under one interface?

Reply:
I actually prefer keeping it like it is, for simplicity